### PR TITLE
feat: flush or compact table and region functions

### PR DESCRIPTION
**Pull request by killme2008**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR implements SQL functions:

* `flush_table(table_name)` to flush a table memtables into SST.
* `compact_table(table_name)` to schedule a compaction task for a table.
* `flush_region(region_id)` to flush a region by the region id.
* `compact_region(region_id)` to schedule a compaction task for the region.

For example:

```sql
mysql> select flush_table("test");
+---------------------------+
| flush_table(Utf8("test")) |
+---------------------------+
|                         0 |
+---------------------------+
1 row in set (0.04 sec)

mysql> select compact_table("test");
+-----------------------------+
| compact_table(Utf8("test")) |
+-----------------------------+
|                           0 |
+-----------------------------+
1 row in set (0.01 sec)

```

Add:

* `admin_fn` attribute macro to simplify the definition of administration functions #3365 
* Refactor `procedure_state` and `migrate_region` function with `admin_fn` macro.

Reduce some tech debts:

* Duplicated `AffectedRows` definitions.
* Move `table_idents_to_full_name` to the session crate etc.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.
- [ ]  This PR does not require documentation updates.

## Refer to a related PR or issue link (optional)

#3161 #2921 #3365 
close https://github.com/GreptimeTeam/greptimedb/issues/2899

### CODE REVIEW

**Comment by killme2008** [comment]
@waynexia @zhongzc @fengjiachun  Please take a look.

File: src/operator/src/request.rs [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+                .partition_manager
+                .find_region_leader(region_id.into())
+                .await
+                .context(FindRegionLeaderSnafu)?;
+
+            inserts.insert(peer, req);
+        }
+
+        Ok(inserts)
+    }
```
Comment (zhongzc) [inline src/operator/src/request.rs:207]: `flush` and `compact` are not grouped here, `Flush` and `Compact` in `RegionRequestBody` do not support batch

Comment (killme2008) [reply src/operator/src/request.rs:207]: Yes, it doesn't do any batch here. The function name is not correct.  Maybe `route_request_by_peer` is better.

**Comment by fengjiachun** [comment]
XXL

File: src/common/function/src/table/flush_compact_region.rs [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+use crate::helper::cast_u64;
+
+macro_rules! define_region_function {
+    ($name: expr, $display_name_str: expr, $display_name: ident) => {
+        /// A function to $display_name
+        #[admin_fn(
+                                                            name = $name,
+                                                            display_name = $display_name_str,
+                                                            sig_fn = "signature",
+                                                            ret = "uint64")]
```
Comment (zhongzc) [inline src/common/function/src/table/flush_compact_region.rs:42]: nit:
```suggestion
        #[admin_fn(name = $name, display_name = $display_name_str, sig_fn = "signature", ret = "uint64")]
```

**Comment by zhongzc** [comment]
rest LGTM

**Comment by killme2008** [comment]
> SIze: XXL 🐶

Don't be afraid. Lots of boilerplate code and tests.

File: src/operator/src/request.rs (hunk 1 above)
Comment (killme2008) [reply src/operator/src/request.rs]: Fixed in https://github.com/GreptimeTeam/greptimedb/pull/3363/commits/3121fa26bf60ba8709b3a1e2bb07559048b2a5db

File: src/common/macro/src/admin_fn.rs [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+                let columns_num = columns.len();
+                let rows_num = if columns.is_empty() {
+                    1
+                } else {
+                    columns[0].len()
+                };
+                let columns = Vec::from(columns);
+
+
```
Comment (waynexia) [inline src/common/macro/src/admin_fn.rs:189]:
```suggestion

```

File: src/common/macro/src/admin_fn.rs [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+                std::thread::spawn(move || {
```
Comment (waynexia) [inline src/common/macro/src/admin_fn.rs:190]: add a comment to this `thread::spawn`. The original comment is not moved to here.

**Review by zhongzc — APPROVED** [review]
LGTM

**Review by waynexia — APPROVED** [review]

**Pull request merged**